contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -0,0 +1,439 @@
+    }
+
+    /**
+     * @notice Update peripheral contract addresses (CurrencyManager, ExecutionManager, TransferSelectorNFT, RoyaltyFeeRegistry)
+     * @dev This function can be called by anyone.
+     */
+    function updatePeripheralContractAddresses() external {
+        currencyManager = looksRareExchange.currencyManager();
+        executionManager = looksRareExchange.executionManager();
+        transferSelectorNFT = ITransferSelectorNFTExtended(address(looksRareExchange.transferSelectorNFT()));
+        IRoyaltyFeeManagerExtended royaltyFeeManager = IRoyaltyFeeManagerExtended(
+            address(looksRareExchange.royaltyFeeManager())
+        );
+        royaltyFeeRegistry = royaltyFeeManager.royaltyFeeRegistry();
+    }
+
+    /**
+     * @notice Check the validity of a maker order
+     * @param makerOrder maker order struct
+     * @return validationCode validation code for the order
+     */
+    function checkOrderValidity(OrderTypes.MakerOrder calldata makerOrder)
+        external
+        view
+        returns (uint256 validationCode)
+    {
+        uint256 response = checkValidityNonces(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        response = checkValidityAmounts(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        response = checkValiditySignature(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        response = checkValidityWhitelists(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        response = checkValidityMinPercentageToAsk(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        response = checkValidityTimestamps(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+        response = checkValidityApprovalsAndBalances(makerOrder);
+        if (response != ORDER_EXPECTED_TO_BE_VALID) return response;
+    }
+
+    /**
+     * @notice Check validity of nonces
+     * @param makerOrder maker order struct
+     * @return validationCode validation code
+     */
+    function checkValidityNonces(OrderTypes.MakerOrder calldata makerOrder)
+        public
+        view
+        returns (uint256 validationCode)
+    {
+        if (looksRareExchange.isUserOrderNonceExecutedOrCancelled(makerOrder.signer, makerOrder.nonce))
+            return NONCE_EXECUTED_OR_CANCELLED;
+        if (makerOrder.nonce < looksRareExchange.userMinOrderNonce(makerOrder.signer))
+            return NONCE_BELOW_MIN_ORDER_NONCE;
+    }
+
+    /**
+     * @notice Check validity of amounts
+     * @param makerOrder maker order struct
+     * @return validationCode validation code
+     */
+    function checkValidityAmounts(OrderTypes.MakerOrder calldata makerOrder)
+        public
+        pure
+        returns (uint256 validationCode)
+    {
+        if (makerOrder.amount == 0) return ORDER_AMOUNT_CANNOT_BE_NULL;
```

Comment:
Maybe we should say `ORDER_AMOUNT_CANNOT_BE_ZERO`?

Reply:
Indeed, it is better! Adjusted.